buildSrc/src/test/java/org/opensearch/gradle/pluginzip/PublishTests.java

```diff
@@ -271,6 +271,76 @@ public void useDefaultValues() throws IOException, URISyntaxException, XmlPullPa
         assertEquals(model.getUrl(), "https://github.com/doe/sample-plugin");
     }
 
+    /**
+     * If the `group` is defined in gradle's allprojects section then it does not have to defined in publications.
+     */
+    @Test
+    public void allProjectsGroup() throws IOException, URISyntaxException, XmlPullParserException {
+        GradleRunner runner = prepareGradleRunnerFromTemplate("allProjectsGroup.gradle", "build", ZIP_PUBLISH_TASK);
+        BuildResult result = runner.build();
+
+        /** Check if build and {@value ZIP_PUBLISH_TASK} tasks have run well */
+        assertEquals(SUCCESS, result.task(":" + "build").getOutcome());
+        assertEquals(SUCCESS, result.task(":" + ZIP_PUBLISH_TASK).getOutcome());
+
+        // Parse the maven file and validate default values
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        Model model = reader.read(
+            new FileReader(
+                new File(
+                    projectDir.getRoot(),
+                    String.join(
+                        File.separator,
+                        "build",
+                        "local-staging-repo",
+                        "org",
+                        "opensearch",
+                        PROJECT_NAME,
+                        "2.0.0.0",
+                        PROJECT_NAME + "-2.0.0.0.pom"
+                    )
+                )
+            )
+        );
+        assertEquals(model.getVersion(), "2.0.0.0");
+        assertEquals(model.getGroupId(), "org.opensearch");
+    }
+
+    /**
+     * The groupId value can be defined on several levels. This tests that the most internal level outweighs other levels.
+     */
```

Comment:
Hey @lukas-vlcek quick question, does this test when `groupId` is added to the exact publication level? 
Example as 
```
publishing {
    publications {
        pluginZip(MavenPublication) { publication ->
            pom {
                name = "opensearch-job-scheduler"
                description = "OpenSearch Job Scheduler plugin"
                groupId = "org.opensearch.plugin"
            }
        }
    }
}
```
`This tests that the most internal level outweighs other levels.` ---> Meaning the `groupId` value when added at the publication level will take more precedence?

Reply:
@prudhvigodithi The answer is in the `groupPriorityLevel.gradle` file (which this test is using). And as you can see the file looks like this:
```gradle
allprojects {
  group = 'level.1'   // Level 1.
}

publishing {
  publications {
    pluginZip(MavenPublication) { publication ->
      groupId = "level.2"   // Level 2.
      pom {
        name = "sample-plugin"
        description = "pluginDescription"
        groupId = "level.3"   // Level 3.
      }
    }
  }
}
```
In which case the **Level 3.** wins.

Reply:
Also you can see that the POM file is physically deployed into `<...>/level/3/<...>` folder (because it is following the groupId).

Reply:
Thanks @lukas-vlcek